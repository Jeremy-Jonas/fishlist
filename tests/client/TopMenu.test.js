import React from 'react'
import {shallow} from 'enzyme'

import TopMenu from '../../client/components/TopMenu'

describe('TopMenu', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TopMenu debug />)

    expect(component).toMatchSnapshot()
  })
})

it('should render correctly with no props', () => {
  const component = shallow(<TopMenu />)

  expect(component).toMatchSnapshot()
})
