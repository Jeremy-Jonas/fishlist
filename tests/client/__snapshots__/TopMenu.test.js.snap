// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMenu should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopMenu
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "borderless": true,
      "children": <Container>
        <MenuItem
          as={[Function]}
          header={true}
          to="/"
        >
          <Header
            as="h1"
            inverted={true}
          >
            The Fish List
          </Header>
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/"
        >
          Home
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/new"
        >
          Add Species
        </MenuItem>
        <Connect(LoginForm) />
      </Container>,
      "color": "green",
      "fixed": "top",
      "inverted": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            as={[Function]}
            header={true}
            to="/"
          >
            <Header
              as="h1"
              inverted={true}
            >
              The Fish List
            </Header>
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/"
          >
            Home
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/new"
          >
            Add Species
          </MenuItem>,
          <Connect(LoginForm) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": <Header
              as="h1"
              inverted={true}
            >
              The Fish List
            </Header>,
            "header": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h1",
              "children": "The Fish List",
              "inverted": true,
            },
            "ref": null,
            "rendered": "The Fish List",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Add Species",
            "to": "/new",
          },
          "ref": null,
          "rendered": "Add Species",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "borderless": true,
        "children": <Container>
          <MenuItem
            as={[Function]}
            header={true}
            to="/"
          >
            <Header
              as="h1"
              inverted={true}
            >
              The Fish List
            </Header>
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/"
          >
            Home
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/new"
          >
            Add Species
          </MenuItem>
          <Connect(LoginForm) />
        </Container>,
        "color": "green",
        "fixed": "top",
        "inverted": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              as={[Function]}
              header={true}
              to="/"
            >
              <Header
                as="h1"
                inverted={true}
              >
                The Fish List
              </Header>
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/"
            >
              Home
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/new"
            >
              Add Species
            </MenuItem>,
            <Connect(LoginForm) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": <Header
                as="h1"
                inverted={true}
              >
                The Fish List
              </Header>,
              "header": true,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h1",
                "children": "The Fish List",
                "inverted": true,
              },
              "ref": null,
              "rendered": "The Fish List",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Add Species",
              "to": "/new",
            },
            "ref": null,
            "rendered": "Add Species",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "borderless": true,
      "children": <Container>
        <MenuItem
          as={[Function]}
          header={true}
          to="/"
        >
          <Header
            as="h1"
            inverted={true}
          >
            The Fish List
          </Header>
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/"
        >
          Home
        </MenuItem>
        <MenuItem
          as={[Function]}
          to="/new"
        >
          Add Species
        </MenuItem>
        <Connect(LoginForm) />
      </Container>,
      "color": "green",
      "fixed": "top",
      "inverted": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            as={[Function]}
            header={true}
            to="/"
          >
            <Header
              as="h1"
              inverted={true}
            >
              The Fish List
            </Header>
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/"
          >
            Home
          </MenuItem>,
          <MenuItem
            as={[Function]}
            to="/new"
          >
            Add Species
          </MenuItem>,
          <Connect(LoginForm) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": <Header
              as="h1"
              inverted={true}
            >
              The Fish List
            </Header>,
            "header": true,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h1",
              "children": "The Fish List",
              "inverted": true,
            },
            "ref": null,
            "rendered": "The Fish List",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": [Function],
            "children": "Add Species",
            "to": "/new",
          },
          "ref": null,
          "rendered": "Add Species",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "borderless": true,
        "children": <Container>
          <MenuItem
            as={[Function]}
            header={true}
            to="/"
          >
            <Header
              as="h1"
              inverted={true}
            >
              The Fish List
            </Header>
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/"
          >
            Home
          </MenuItem>
          <MenuItem
            as={[Function]}
            to="/new"
          >
            Add Species
          </MenuItem>
          <Connect(LoginForm) />
        </Container>,
        "color": "green",
        "fixed": "top",
        "inverted": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              as={[Function]}
              header={true}
              to="/"
            >
              <Header
                as="h1"
                inverted={true}
              >
                The Fish List
              </Header>
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/"
            >
              Home
            </MenuItem>,
            <MenuItem
              as={[Function]}
              to="/new"
            >
              Add Species
            </MenuItem>,
            <Connect(LoginForm) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": <Header
                as="h1"
                inverted={true}
              >
                The Fish List
              </Header>,
              "header": true,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "h1",
                "children": "The Fish List",
                "inverted": true,
              },
              "ref": null,
              "rendered": "The Fish List",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": "Add Species",
              "to": "/new",
            },
            "ref": null,
            "rendered": "Add Species",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
